staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -184,7 +184,6 @@ func (i *indexedWatchers) terminateAll(groupResource schema.GroupResource, done
 // second in a bucket, and pop up them once at the timeout. To be more specific,
 // if you set fire time at X, you can get the bookmark within (X-1,X+1) period.
 type watcherBookmarkTimeBuckets struct {
```

Comment:
it looks like the `watcherBookmarkTimeBuckets` is called from only 3 method: `startDispatching`, `finishDispatching` and `Watch`. All these methods acquire `c.Lock()` before touching `watcherBookmarkTimeBuckets`.

thus we could remove explicit locking in some methods of `watcherBookmarkTimeBuckets` since the access is already synced.

Reply:
I think this would simplify https://github.com/kubernetes/kubernetes/pull/117014 (we could remove the refactoring)

Reply:
@wojtek-t wdyt?

Reply:
This makes sense to me (especially that other watchers maps are also lock-free and protected by higher-level lock).

So yeah - let's make this happen.

Reply:
done, PTAL.